File: app/src/main/java/com/neome/java/api/meta/base/dto/StudioEntAutomationStepGenerateDocument.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.MetaIdGrid;
import com.neome.java.api.meta.base.Types.MetaIdLayoutForm;
import com.neome.java.api.meta.base.Types.MetaIdPipelineParam;
import com.neome.java.api.meta.base.Types.MetaIdVar;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class StudioEntAutomationStepGenerateDocument extends StudioEntAutomationStep
{
  @Nullable
  public MetaIdLayoutForm contentLayoutId;

  @Nullable
  public StudioValueVarIdText fileNameVarId;

  @Nullable
  public MetaIdVar inputFormMappingVarId;

  @Nullable
  public MetaIdPipelineParam inputFormPipelineVarId;

  @Nullable
  public StudioValueVarIdCondition iterateOnGridFilterVarId;

  @Nullable
  public MetaIdGrid iterateOnGridId;

  @Nullable
  public MetaIdPipelineParam outputFormPipelineVarId;

  @Nullable
  public MetaIdLayoutForm templateLayoutId;
}

File: app/src/main/java/com/neome/api/meta/base/dto/StudioEntAutomationStepGenerateDocument.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.MetaIdGrid
import com.neome.api.meta.base.Types.MetaIdLayoutForm
import com.neome.api.meta.base.Types.MetaIdPipelineParam
import com.neome.api.meta.base.Types.MetaIdVar
import com.neome.api.meta.base.dto.StudioEntAutomationStep
import com.neome.api.meta.base.dto.StudioValueVarIdCondition
import com.neome.api.meta.base.dto.StudioValueVarIdText

interface StudioEntAutomationStepGenerateDocument : StudioEntAutomationStep
{
  val contentLayoutId: MetaIdLayoutForm?
  val fileNameVarId: StudioValueVarIdText?
  val inputFormMappingVarId: MetaIdVar?
  val inputFormPipelineVarId: MetaIdPipelineParam?
  val iterateOnGridFilterVarId: StudioValueVarIdCondition?
  val iterateOnGridId: MetaIdGrid?
  val outputFormPipelineVarId: MetaIdPipelineParam?
  val templateLayoutId: MetaIdLayoutForm?
}